File: app/src/main/java/com/example/bloodbank/Adapter/BadgeAdapter.java
package com.example.bloodbank.Adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.bloodbank.R;

import java.util.ArrayList;
import java.util.List;

public class BadgeAdapter extends RecyclerView.Adapter<BadgeAdapter.BadgeViewHolder> {
    private List<BadgeItem> badges = new ArrayList<>();

    public static class BadgeItem {
        private String name;
        private String description;
        private int iconResId;
        private boolean isUnlocked;

        public BadgeItem(String name, String description, int iconResId, boolean isUnlocked) {
            this.name = name;
            this.description = description;
            this.iconResId = iconResId;
            this.isUnlocked = isUnlocked;
        }

        public String getName() { return name; }
        public String getDescription() { return description; }
        public int getIconResId() { return iconResId; }
        public boolean isUnlocked() { return isUnlocked; }
    }

    public static class BadgeViewHolder extends RecyclerView.ViewHolder {
        private ImageView badgeIcon;
        private TextView badgeName;
        private TextView badgeDescription;
        private View lockOverlay;

        public BadgeViewHolder(@NonNull View itemView) {
            super(itemView);
            badgeIcon = itemView.findViewById(R.id.badgeIcon);
            badgeName = itemView.findViewById(R.id.badgeName);
            badgeDescription = itemView.findViewById(R.id.badgeDescription);
            lockOverlay = itemView.findViewById(R.id.lockOverlay);
        }

        public void bind(BadgeItem badge) {
            badgeIcon.setImageResource(badge.getIconResId());
            badgeName.setText(badge.getName());
            badgeDescription.setText(badge.getDescription());
            
            if (badge.isUnlocked()) {
                lockOverlay.setVisibility(View.GONE);
                itemView.setAlpha(1.0f);
            } else {
                lockOverlay.setVisibility(View.VISIBLE);
                itemView.setAlpha(0.5f);
            }
        }
    }

    @NonNull
    @Override
    public BadgeViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_badge, parent, false);
        return new BadgeViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull BadgeViewHolder holder, int position) {
        holder.bind(badges.get(position));
    }

    @Override
    public int getItemCount() {
        return badges.size();
    }

    public void setBadges(List<BadgeItem> badges) {
        this.badges = badges;
        notifyDataSetChanged();
    }
} 

File: app/src/main/java/com/example/bloodbank/Adapter/BadgeAdapter.kt
package com.example.bloodbank.Adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.bloodbank.R

/**
 * BadgeAdapter - RecyclerView adapter for displaying earned badges.
 * Kotlin version with horizontal layout support.
 */
class BadgeAdapter(private var badges: List<String> = emptyList()) : 
    RecyclerView.Adapter<BadgeAdapter.BadgeViewHolder>() {
    
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BadgeViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_badge, parent, false)
        return BadgeViewHolder(view)
    }
    
    override fun onBindViewHolder(holder: BadgeViewHolder, position: Int) {
        holder.bind(badges[position])
    }
    
    override fun getItemCount(): Int = badges.size
    
    /**
     * Update badges list and refresh UI.
     */
    fun updateBadges(newBadges: List<String>) {
        badges = newBadges
        notifyDataSetChanged()
    }
    
    inner class BadgeViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val badgeIcon: ImageView = itemView.findViewById(R.id.badgeIcon)
        private val badgeNameText: TextView = itemView.findViewById(R.id.badgeNameText)
        
        fun bind(badgeName: String) {
            badgeNameText.text = badgeName
            
            // Use Android's built-in star icon for all badges
            // TODO: Add custom badge icons to res/drawable
            badgeIcon.setImageResource(android.R.drawable.btn_star_big_on)
        }
    }
}
